### ISSUE TITLE
Set up figshare for storing data and models

### ISSUE THREAD

**Issue by nashirj** [ucfai/knightros-gambit#151]
Use https://figshare.com/ to store our models and data. It's what sklearn uses, and we can set up something similar to how they fetch datasets: https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/datasets/_california_housing.py#L53

**Comment by ryPattillo** [comment]
One of us needs to create an account. We can do that at the meeting tonight.

**Comment by fshcat** [comment]
just making sure, do we want the code to automatically write to the figshare or just read from it and we just manually upload new dataset if necessary

**Comment by nashirj** [comment]
> just making sure, do we want the code to automatically write to the figshare or just read from it and we just manually upload new dataset if necessary

This is a good question and honestly I'm not sure. I think we should discuss this more at tomorrow's software meeting. I think it depends on if we are iteratively building the dataset or we are just going to build a few different datasets then merge them in a separate step to train on with mcts

**Comment by nashirj** [comment]
We just talked about this and there are two options:
1) Build dataset all at once: this would mean we don't need to overwrite, can just download and merge and do mcts training
2) Iteratively go back and forth: this would allow incremental model training, but it adds a lot of complexity with file writing/reading/merging.

**decision**: create dataset all at once; download and train after as necessary

**Comment by nashirj** [comment]
tl;dr: We need to add logic in `train.py` for
1) Writing data to figshare after creating it in `create_dataset` (we can do a proof of concept in experimental work)
2) Downloading data from figshare in the event that we want to use existing data (need to update hyperparameters file to use figshare user path and fetch all datasets from user)
3) Merging downloaded data from figshare

This can be done in one PR

**Comment by ryPattillo** [comment]
I am working on this right now ^ . Should we have something in params.json that specifies uploading to figshare? Or we always going to upload to figshare after creating the dataset?

**Comment by ryPattillo** [comment]
Same with models

**Comment by ryPattillo** [comment]
Adding a figshare boolean to params.json

**Comment by ryPattillo** [comment]
This has been completed.
